tools/travis-test.sh

```diff
@@ -48,7 +48,10 @@ setup_base()
   if [ -z "$USE_DEBUG" ]; then
     $PIP install -v . 2>&1 | tee log
   else
-    # Python3.5-dbg on travis seems to need this
+    pip install --upgrade setuptools toml
+    # Install the build dependencies from pyproject.toml.
+    pip install `python -c "import toml; req = toml.load('pyproject.toml')['build-system']['requires']; print(' '.join(req))"`
```

Comment:
This is quite weird, overall it's less readable. Why do this?

Reply:
It installs the build dependencies from `pyproject.toml`.  That's where the list of dependencies now resides, correct?  (See https://github.com/numpy/numpy/pull/15275#discussion_r364186088)

It turns out I'm still iterating on this PR, so expect this to change shortly.  I'll still use this code snippet, but not here.

Reply:
pyproject.toml, which is a thing still in flux since there is not even an official toml reader in the stdlib, holds the build dependencies. The test_requirements.txt holds the test dependencies. I think for CI we should stick with the test dependencies and not mess with the toml file.

Reply:
using only the pyproject.toml will break dependabot.

Reply:
Which files does dependabot read?

Reply:
Partially answering my own question: dependabot looks at `test_requirements.txt`.  Does it read anything else?